Avalonia.Lottie/Animation/Content/LottieCanvas.cs

```diff
@@ -11,37 +12,86 @@ namespace Avalonia.Lottie.Animation.Content
 {
     public class LottieCanvas
     {
-        private readonly Stack<RenderTargetSave> _renderTargetSaves = new();
-
         public LottieCanvas(double width, double height)
         {
             UpdateSize(width, height);
         }
 
-        private DrawingContext CurrentDrawingContext =>
-            _renderTargetSaves.Count > 0 ? _renderTargetSaves.Peek().Context : null;
+        private uint _layerIndex;
+        private Dictionary<uint, RenderTargetSave> _layerCache = new();
+        private Size _lastSize = Size.Empty;
+
+        private IDrawingContextImpl mainDrawingContext;
+
+
+        private DrawingContext CurrentDrawingContext => ContextStack.Peek();
+
+        private DrawingContext _layerCtx;
+        private Stack<DrawingContext> ContextStack = new();
 
         public double Width { get; private set; }
         public double Height { get; private set; }
 
         private void UpdateSize(double width, double height)
         {
+            if (double.IsNaN(width) || double.IsNaN(height))
+            {
+                return;
+            }
+            
             Width = width;
             Height = height;
+            var curSize = new Size(Width, Height);
+
+            if (curSize != _lastSize)
+            {
+                _lastSize = curSize;
+                InvalidateLayerCache();
+            }
         }
 
-        internal IDisposable CreateSession(Size size, IDrawingContextLayerImpl layer,
-            DrawingContext drawingSession)
+        private void InvalidateLayerCache()
         {
-            UpdateSize(size.Width, size.Height);
+            if(_layerCache.Count == 0) return;
+            
+            foreach (var rts in _layerCache.Values)
+            {
+                rts.Dispose();
+            }
 
-            var rts = new RenderTargetSave(layer,
-                drawingSession, size,
-                new PorterDuffXfermode(PorterDuff.Mode.Clear));
+            _layerCache.Clear();
+        }
 
-            _renderTargetSaves.Push(rts);
+        private RenderTargetSave GetOrCreateLayer(uint layerIndex, Rect bounds, Paint paint)
+        {
+            // if (_layerCache.TryGetValue(layerIndex, out var output))
+            // {
+            //     return output;
+            // }
 
-            return new Disposable(() => { _renderTargetSaves.Clear(); });
+
+            
+            var renderTarget = mainDrawingContext.CreateLayer(bounds.Size);
+
+            var rts = new RenderTargetSave(renderTarget,
+                new DrawingContext(renderTarget.CreateDrawingContext(null)),
```

Comment:
This is causing deadlocks on windows (and probably other problems on other OSes). You are creating this crawing context and RT context but it is never disposed.

Reply:
@MarchingCube 🥇 you deserve a freaking medal mate :D :D finally fixed!!!!!!!